pkgs/build-support/node/fetch-yarn-deps/index.js

```diff
@@ -143,8 +143,17 @@ const performParallel = tasks => {
 
 const prefetchYarnDeps = async (lockContents, verbose) => {
 	const lockData = lockfile.parse(lockContents)
+	// This could be implemented using [`Map.groupBy`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/groupBy),
+	// but that method is only supported starting with Node 21
+	const uniqueBy = (arr, callback) => {
+		const map = new Map()
+		for (const elem of arr) {
+			map.set(callback(elem), elem)
+		}
+		return [...map.values()]
+	}
```

Comment:
Could you please define this function outside the `const prefetchYarnDeps` function - it should improve performance.

Reply:
Yeah makes sense, done